Implement AssertJ representation

I've submitted two PRs, not sure which approach is preferred. In both cases, an exception message for matching against the wrong type of Arg goes from:

```
java.lang.AssertionError: 
Expecting:
 <[test, version]>
to contain:
 <[test, version]>
but could not find:
 <[test]>
```

to

```
java.lang.AssertionError: 
Expecting:
 <[SafeArg[id=test], SafeArg[version=version]]>
to contain:
 <[UnsafeArg[id=test], SafeArg[version=version]]>
but could not find:
 <[UnsafeArg[id=test]]>
```

Happy to add tests to whichever approach is considered better.

--- a/preconditions-assertj/src/main/java/com/palantir/logsafe/testing/LoggableArgRepresentation.java
+++ b/preconditions-assertj/src/main/java/com/palantir/logsafe/testing/LoggableArgRepresentation.java
@@ -28,7 +28,7 @@ public class LoggableArgRepresentation extends StandardRepresentation  {
             return String.format("%s[%s=%s]",
                     arg.getClass().getSimpleName(),
                     arg.getName(),
-                    super.fallbackToStringOf(arg.getValue()));
+                    toStringOf(arg.getValue()));
         }
         return super.fallbackToStringOf(object);
     }

--- a/preconditions-assertj/src/main/java/com/palantir/logsafe/testing/LoggableArgRepresentation.java
+++ b/preconditions-assertj/src/main/java/com/palantir/logsafe/testing/LoggableArgRepresentation.java
@@ -1,0 +1,36 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.logsafe.testing;
+
+import com.palantir.logsafe.Arg;
+import org.assertj.core.presentation.StandardRepresentation;
+
+public class LoggableArgRepresentation extends StandardRepresentation  {
+
+    @Override
+    protected final String fallbackToStringOf(Object object) {
+        if (object instanceof Arg) {
+            Arg<?> arg = (Arg<?>) object;
+            return String.format("%s[%s=%s]",
+                    arg.getClass().getSimpleName(),
+                    arg.getName(),
+                    super.fallbackToStringOf(arg.getValue()));
+        }
+        return super.fallbackToStringOf(object);
+    }
+
+}

--- a/preconditions-assertj/src/main/resources/META-INF/services/org.assertj.core.presentation.Representation
+++ b/preconditions-assertj/src/main/resources/META-INF/services/org.assertj.core.presentation.Representation
@@ -1,0 +1,1 @@
+com.palantir.logsafe.testing.LoggableArgRepresentation

--- a/preconditions-assertj/src/test/java/com/palantir/logsafe/testing/LoggableArgRepresentationTest.java
+++ b/preconditions-assertj/src/test/java/com/palantir/logsafe/testing/LoggableArgRepresentationTest.java
@@ -1,0 +1,50 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.logsafe.testing;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.UnsafeArg;
+import java.util.concurrent.CompletableFuture;
+import org.junit.Test;
+
+public class LoggableArgRepresentationTest {
+
+    private final LoggableArgRepresentation representation = new LoggableArgRepresentation();
+
+    @Test
+    public void toStringOf_safeArg() {
+        assertThat(representation.toStringOf(SafeArg.of("hello", "world")))
+                .isEqualTo("SafeArg[hello=world]");
+    }
+
+    @Test
+    public void toStringOf_safeArg_complexType() {
+        CompletableFuture<String> stringFuture = new CompletableFuture<>();
+        assertThat(representation.toStringOf(stringFuture)).isEqualTo("CompletableFuture[Incomplete]");
+        assertThat(representation.toStringOf(SafeArg.of("future", stringFuture)))
+                .isEqualTo("SafeArg[future=CompletableFuture[Incomplete]]");
+    }
+
+    @Test
+    public void toStringOf_unsafeArg() {
+        assertThat(representation.toStringOf(UnsafeArg.of("hello", "world")))
+                .isEqualTo("UnsafeArg[hello=world]");
+    }
+
+}

--- a/preconditions-assertj/src/test/java/com/palantir/logsafe/testing/LoggableArgRepresentationTest.java
+++ b/preconditions-assertj/src/test/java/com/palantir/logsafe/testing/LoggableArgRepresentationTest.java
@@ -30,7 +30,7 @@ public class LoggableArgRepresentationTest {
     @Test
     public void toStringOf_safeArg() {
         assertThat(representation.toStringOf(SafeArg.of("hello", "world")))
-                .isEqualTo("SafeArg[hello=world]");
+                .isEqualTo("SafeArg[hello=\"world\"]");
     }
 
     @Test
@@ -44,7 +44,7 @@ public class LoggableArgRepresentationTest {
     @Test
     public void toStringOf_unsafeArg() {
         assertThat(representation.toStringOf(UnsafeArg.of("hello", "world")))
-                .isEqualTo("UnsafeArg[hello=world]");
+                .isEqualTo("UnsafeArg[hello=\"world\"]");
     }
 
 }

